onnxruntime/test/providers/base_tester.cc

```diff
@@ -41,6 +41,14 @@ void DebugTrap() {
 #endif
 }  // namespace
 
+static inline bool use_cosine_similarity() {
+  static auto value = [&] {
+    const char* ptr = std::getenv("ORT_TEST_USE_COSINE_SIMILARITY");
```

Comment:
I am not sure about test environment setup, but historically ORT does not use environment variables as switches. Adding this would make the testing pipeline more complex, causing either flaky tests or missed unit testing.

Reply:
removed all the env variables and used the the existing arguments to decide on cosine similarity checks.